module/Finna/src/Finna/RecordDriver/SolrDefault.php

```diff
@@ -62,4 +62,21 @@ public function __construct(
         parent::__construct($mainConfig, $recordConfig, $searchSettings);
         $this->searchSettings = $searchSettings;
     }
+
+    /**
+     * Return the collection search ID for this record.
+     *
+     * @return string
+     */
+    public function getCollectionSearchId(): string
+    {
+        if ($this->mainConfig->Hierarchy->showFullHierarchyTree) {
```

Comment:
Ja tässä pitäisi olla joku tarkastus ettei nullista yritä katsoa arvoa,
esim $this->mainConfig->Hierarchy->showFullHierarchyTree ?? false